server/src/main/java/com/hedera/block/server/persistence/storage/read/BlockAsDirReaderBuilder.java

```diff
@@ -16,37 +16,40 @@
 
 package com.hedera.block.server.persistence.storage.read;
 
-import com.hedera.block.server.persistence.storage.FileUtils;
+import static com.hedera.block.common.utils.FileUtilities.DEFAULT_DIR_PERMISSIONS;
```

Comment:
Agreed about that.

Reply:
1. Yes, both reader and writer are not expected to be used in production, but we mustn't design them as test code, because any operator _might_ choose to use them as the production storage for whatever reason (perhaps their node stores a small subset of block data and they want other processes to have easy access).
1. Always try to design software well, and make the formal design documents (whether Java, C++, Rust, etc...) as readable and clear as you can within constraints.
   * This includes good practice of configurable classes with reasonable defaults managed as private data within the class.
1. It is reasonable to remove the filePerms field, but might be _better_ to leave it in with a default value of `null`.

Reply:
✅ Done.